staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -397,11 +383,16 @@ func (wc *watchChan) startWatching(watchClosedCh chan struct{}, initialEventsEnd
 			err := wres.Err()
 			// If there is an error on server (e.g. compaction), the channel will return it before closed.
 			logWatchChannelErr(err)
+			// sendError doesn't guarantee that no more items will be put into resultChan.
+			// However, by returning from startWatching here, we guarantee, that events
+			// with higher resourceVersion than the error will not be queue and thus also
+			// processed and send to the user.
+			// TODO(wojtek-t): Figure out if we can synchronously prevent more events.
```

Comment:
maybe just move this comment (and the one below) to the sendError function? we have 7 places we call it... probably don't need this comment on the call sites

Reply:
For other calls, the comment is non-relevant, because they actually break and ensure that nothing else is put into resultChan.
The difference is that here we only put things to incomingCh channel and we process them in a separate goroutine.